File: milliampere_dp/src/milliampere_dp/rendering/colors.py
"""Color constants for milliAmpere1 vessel rendering.

Extracted from dashboard.py Color enum. RGB tuples for opaque colors,
RGBA tuples where alpha is needed.
"""

from __future__ import annotations

from enum import Enum


class Color(Enum):
    """Color constants for vessel rendering."""

    WHITE = (255, 255, 255)
    GRAY = (150, 150, 150)
    BLACK = (0, 0, 0)
    GREEN = (0, 255, 0)
    PURPLE = (255, 0, 255)
    RED = (255, 0, 0)
    ORANGE = (255, 128, 0)

    SCREEN_COLOR = (240, 240, 240)

    TABLEAU_BLUE = (87, 120, 164)
    TABLEAU_ORANGE = (228, 148, 68)
    TABLEAU_GREEN = (106, 159, 88)
    TABLEAU_RED = (209, 97, 93)

    LEGEND_BOX = (200, 200, 200, 180)

    AGENT_BLACK = (0, 0, 0, 0)
    AGENT_BLUE = (0, 122, 255, 220)
    DESIRED_YELLOW = (255, 102, 0)
    DESIRED_LIGHT_YELLOW = (255, 205, 128)
    OCEAN_BLUE = (209, 237, 255)
    OCEAN_GRID = (182, 206, 222)
    VELOCITY_GREEN = (0, 255, 123)

    LIGHT_RED = (255, 171, 171)

    @property
    def hex(self) -> str:
        """Web-compatible hex color string (e.g. '#ff0000').

        Uses only the first 3 channels (ignores alpha for RGBA colors).
        """
        r, g, b = self.value[0], self.value[1], self.value[2]
        return f"#{r:02x}{g:02x}{b:02x}"

File: web/src/colors.ts
/**
 * Color constants matching milliampere_dp/rendering/colors.py.
 *
 * All values as CSS hex strings for Canvas 2D.
 */

export const Colors = {
  WHITE: "#ffffff",
  GRAY: "#969696",
  BLACK: "#000000",
  GREEN: "#00ff00",
  PURPLE: "#ff00ff",
  RED: "#ff0000",
  ORANGE: "#ff8000",

  SCREEN_COLOR: "#f0f0f0",

  TABLEAU_BLUE: "#5778a4",
  TABLEAU_ORANGE: "#e49444",
  TABLEAU_GREEN: "#6a9f58",
  TABLEAU_RED: "#d1615d",

  LEGEND_BOX: "rgba(200, 200, 200, 0.71)",

  AGENT_BLUE: "rgba(0, 122, 255, 0.86)",
  DESIRED_YELLOW: "#ff6600",
  DESIRED_LIGHT_YELLOW: "#ffcd80",
  OCEAN_BLUE: "#d1edff",
  OCEAN_GRID: "#b6cede",
  VELOCITY_GREEN: "#00ff7b",

  LIGHT_RED: "#ffabab",
} as const;
